algorithms/dp/bitmasking_tsp.py

```diff
@@ -1,15 +1,187 @@
+"""
+def check_argument(nodes: list, nbRow: int, nbColumn: int):
+    if len(nodes) != nbRow:
+        raise ValueError(f'The number of rows in nodes ({len(nodes)}) does not match the given number of row: {nbRow} !')
+    for row in nodes:
+        if not isinstance(row, list):
+            raise ValueError(f'The given row {row} is not valid !')
+        if len(row) != nbColumn:
+            raise ValueError(f'The number of columns in row {row} does not match the given number of column: {nbColumn} !')
+        for node in row:
+            if node != '.' and node != '*' and node != '#':
+                raise ValueError(f'The given node "{node}" in row {row} is not valid !')
+
+"""
+Check if the given position is valid
+
+:param nodes: The nodes
+:param nbRow: The number of rows
+:param nbColumn: The number of columns
+:returns: True if the position is valid, False otherwise
+"""
+def is_safe_pos(nodes: list, nbRow: int, nbColumn: int, r: int, c: int):
+    return r >= 0 and r < nbRow and c >= 0 and c < nbColumn and nodes[r][c] != '#'
+
+
+"""
+Compute all the distances from the given house to all nodes using BFS
+
+:param nodes: The nodes
+:param houses: The houses location
+:param nbRow: The number of rows
+:param nbColumn: The number of columns
+:param dist: All distances
+:param index: The house index
+:returns: Nothing 
+"""
+def getDist(nodes: list, houses: list, nbRow: int, nbColumn: int, dist: list, index: int):
+    
+    visited = [[False for c in range(nbColumn)] for r in range(nbRow)]
+
+    curr_row = houses[index][0]
+    curr_col = houses[index][1]
+
+    queue = [[curr_row, curr_col]]
+    visited[curr_row][curr_col] = True
+    dist[curr_row][curr_col][index] = 0
+
+    while queue != []:
+        pos = queue.pop(0)
+        # iterate on all 4 neighbors
+        for dir in range(4):
+            curr_row = pos[0] + D_X[dir]
+            curr_col = pos[1] + D_Y[dir]
+            # check if the neighbor is a valid node
+            if is_safe_pos(nodes, nbRow, nbColumn, curr_row, curr_col):
+                if not visited[curr_row][curr_col]:
+                    visited[curr_row][curr_col] = True
+                    dist[curr_row][curr_col][index] = dist[pos[0]][pos[1]][index] + 1
+                    queue.append([curr_row, curr_col])
+
+"""
+Compute all the distances from all houses to all nodes
+
+:param nodes: The nodes
+:param houses: The houses location
+:param nbRow: The number of rows
+:param nbColumn: The number of columns
+:returns: All the distances
+"""
+def getAllDist(nodes: list, houses: list, nbRow: int, nbColumn: int):
+
+    dist = [[[math.inf for house in range(len(houses))] for column in range(nbColumn)] for row in range(nbRow)]
+
+    for index in range(len(houses)):
+        getDist(nodes, houses, nbRow, nbColumn, dist, index)
+
+    return dist
+
+"""
+Initialize the variables
+
+:param nodes: The nodes given
+:param nbRow: The number of rows
+:param nbColumn: The number of columns
+"""
```

Comment:
Could you add a description of the return statement to the comment?

Reply:
I'll do it, thank you for catching this miss